### PR TITLE
Fix store interfaces

### PR DESCRIPTION
**Pull request by ohadsahar**

### CODE REVIEW

File: core/components/Alert/Alert.tsx [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { Strings } from "@/constants/strings";
 import { AlertProps } from "interfaces/alert.interface";
 import { AlertParagraph } from "./styled";
 
-const Alert = ({ alert }: AlertProps) => {
+const Alert = (alert: AlertProps) => {
```
Comment (amitmarko) [inline core/components/Alert/Alert.tsx]:
```suggestion
const Alert = ({id, msg, alertType}: AlertProps) => {
```

File: core/components/CompletedTasksList/CompletedTasksList.tsx [hunk 2]
```diff
@@ -42,27 +42,26 @@ const CompletedTasksList = ({
...
-                ) : (
-                  <TaskForm task={task} />
-                )}
-              </div>
-            ) : null
+          {tasks?.map(
+            (task: TaskProps) =>
+              task.status === TaskStatusType.COMPLETED && (
+                <div key={task?.id}>
+                  {!task?.editable ? (
```
Comment (amitmarko) [inline core/components/CompletedTasksList/CompletedTasksList.tsx]:
```suggestion
                  {!task?.editMode ? (
```

File: core/components/CreatedTasksList/CreatedTasksList.tsx [hunk 3]
```diff
@@ -21,18 +21,14 @@ const CreatedTasksList = ({
...
-              !task?.editable ? (
-                <Task
-                  task={task}
-                  handleAction={(data: string) => handleAction(data, task)}
-                  handleEdit={() => handleEdit(task)}
-                />
-              ) : (
-                <TaskForm task={task} />
-              )
+            {task.status === TaskStatusType.CREATED && !task?.editable ? (
```
Comment (amitmarko) [inline core/components/CreatedTasksList/CreatedTasksList.tsx]:
```suggestion
            {task.status === TaskStatusType.CREATED && !task?.editMode ? (
```

File: core/components/TaskForm/hooks/useSubmitForm.ts [hunk 4]
```diff
@@ -1,18 +1,18 @@
...
   const [taskValue, setTaskValue] = useState<string>(task?.taskName ?? "");
 
-  const onChangeTaskname = (data: string) => {
+  const onChangeTaskname = useCallback((data: string) => {
     setTaskValue(data);
     if (!taskValue && isError) {
       setError(false);
     }
-  };
+  }, []);
```
Comment (amitmarko) [inline core/components/TaskForm/hooks/useSubmitForm.ts]:
```suggestion
  }, [taskValue, isError]);
```

Comment (ohadsahar) [reply core/components/TaskForm/hooks/useSubmitForm.ts]: Done

File: core/components/Alert/Alert.tsx (hunk 1 above)
Comment (ohadsahar) [reply core/components/Alert/Alert.tsx]: Done

File: core/components/CompletedTasksList/CompletedTasksList.tsx (hunk 2 above)
Comment (ohadsahar) [reply core/components/CompletedTasksList/CompletedTasksList.tsx]: Changed!

File: core/components/CreatedTasksList/CreatedTasksList.tsx (hunk 3 above)
Comment (ohadsahar) [reply core/components/CreatedTasksList/CreatedTasksList.tsx]: Changed

File: core/components/CreatedTasksList/CreatedTasksList.tsx [hunk 5]
```diff
@@ -21,22 +21,18 @@ const CreatedTasksList = ({
...
-              !task?.editable ? (
-                <Task
-                  task={task}
-                  handleAction={(data: string) => handleAction(data, task)}
-                  handleEdit={() => handleEdit(task)}
-                />
-              ) : (
-                <TaskForm task={task} />
-              )
+            {task.status === TaskStatusType.CREATED && !task.editMode ? (
```
Comment (amitmarko) [inline core/components/CreatedTasksList/CreatedTasksList.tsx]: please send created tasks as tasks - `tasks : tasks.filter(({status}) => status === TaskStatusType.CREATED)`

**Pull request merged**